lightning/src/ln/msgs.rs

```diff
@@ -1331,12 +1332,13 @@ impl Readable for OnionHopData {
 			let mut cltv_value = HighZeroBytesDroppedVarInt(0u32);
 			let mut short_id: Option<u64> = None;
 			let mut payment_data: Option<FinalOnionHopData> = None;
-			decode_tlv!(&mut rd, {
-				(2, amt),
-				(4, cltv_value)
-			}, {
-				(6, short_id),
-				(8, payment_data)
+			let mut keysend_preimage: Option<PaymentPreimage> = None;
+			decode_tlv_stream!(&mut rd, {
+				(2, amt, required),
+				(4, cltv_value, required),
+				(6, short_id, option),
+				(8, payment_data, option),
+				(5482373484, keysend_preimage, option)
```

Comment:
I guess the TLV-type is motivated by inter-compat with other implementations ? Doesn't seem really bandwidth-efficient...though maybe add a comment to precise it's a not a magic value picked up by our implementation ?

Reply:
Yeah, it was chosen for interoperability. Not sure why this value was chosen to begin with. Added a comment!